mgmtd/mgmt_fe_adapter.c

```diff
@@ -1172,6 +1164,41 @@ static void fe_adapter_handle_get_data(struct mgmt_fe_session_ctx *session,
 		goto done;
 	}
 
+	switch (msg->datastore) {
+	case MGMT_MSG_DATASTORE_CANDIDATE:
+		ds_id = MGMTD_DS_CANDIDATE;
+		break;
+	case MGMT_MSG_DATASTORE_RUNNING:
+		ds_id = MGMTD_DS_RUNNING;
+		break;
+	case MGMT_MSG_DATASTORE_OPERATIONAL:
+		ds_id = MGMTD_DS_OPERATIONAL;
+		break;
+	default:
+		fe_adapter_send_error(session, req_id, false, -EINVAL,
+				      "Unsupported datastore %" PRIu8
+				      " requested from session-id: %" PRIu64,
+				      msg->datastore, session->session_id);
+		goto done;
+	}
+
+	switch (msg->format) {
```

Comment:
This is exactly the code that we don't need -- just use the libyang type.

Reply:
I removed the translation as it's not needed now, but I preserved the validation of the supplied value, so there's not much less LOC here.

Reply:
OK I think you can trick checkpatch.pl then to pass by using a symbolic constant that i undefined

`#if DONT_DEFINE_UNTIL_NEEDED`